### DISCUSSION TITLE
Geometry Service Error

### DISCUSSION THREAD

**Discussion by ufsaa2022** [ansys/pyansys-geometry#1197] [Q&A] [answered]
Hello, 

I'm trying to run Geometry Service with the script below.

`from ansys.geometry.core import launch_modeler_with_geometry_service`
`modeler = launch_modeler_with_geometry_service()`

But immediately the error below occurs.

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/b5a811da-3a7d-4eaa-afad-1627cdc9400e)

The Python version is 3.10.11, and the PyAnsys Geometry version is 0.5.4.
After creating the virtual environment, I did not install any other libraries other than PyAnsys Geometry.

Is there any solution?

Thanks!

**Comment by RobPasMue** [comment, marked as answer]
> Hi @ufsaa2022 - I see you closed the discussion. Do you still have the problem?

Thread contains resolution of various issues. Marking as answer

**Reply by ufsaa2022** [reply]
Yes, I still have the problem.
I looked at the FAQ in the link below, but it didn't solve the problem.

https://geometry.docs.pyansys.com/version/stable/getting_started/faq.html

**Reply by RobPasMue** [reply]
Let me reopen the discussion in that case. Also, can you verify the following?

1. Do you have the Geometry Service installed? If so, it should be available at the following location:
```C:\Program Files\ANSYS Inc\v241\GeometryService```

**Reply by ufsaa2022** [reply]
It wasn't installed automatically, so I copied and pasted the folder in the *.iso file.

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/967dc0d1-882b-44ea-8eb2-517302b459d9)

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/00b8e73b-5865-4af0-a7b8-615a629e6653)

**Reply by RobPasMue** [reply]
Could you also run the following code?

```py
from ansys.tools.path import get_available_ansys_installations, get_latest_ansys_installation
installs = get_available_ansys_installations()
print(installs)
latest = get_latest_ansys_installation()
print(latest)
```

**Reply by ufsaa2022** [reply]
The output is as follows.

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/dd7b86da-1e57-495e-9d4f-e7133602fda1)

**Reply by RobPasMue** [reply]
Okay good! Can we give it a try now to launching the modeler?

```py
from ansys.geometry.core import launch_modeler_with_geometry_service

modeler = launch_modeler_with_geometry_service()

print(modeler)
```

**Reply by ufsaa2022** [reply]
I get a timeout error.. :(

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/7831a1dc-8f67-44f4-bfb3-5835f6a395c1)

**Reply by RobPasMue** [reply]
That is unfortunate... let's do the following. Can we clear out the ``C:\Program Files\ANSYS Inc\v241\GeometryService\Logs`` folder, and relaunch it? You should get a new file created. Please share it (offuscating any content you want to offuscate like IPs/servers etc.)

I just tried it on my side, and if everything went fine you should get something similar to this:

![image](https://github.com/ansys/pyansys-geometry/assets/37798125/a33f783a-4604-4aae-acd4-aa14a1221fb3)

This is how things looked on my CLI:

![image](https://github.com/ansys/pyansys-geometry/assets/37798125/21482378-26f2-4309-a792-c93b41bbdec4)

**Reply by ufsaa2022** [reply]
The Logs folder does not exist.
Is this because I copied and pasted the Geometry Service instead of installing it?

**Reply by RobPasMue** [reply]
That is unfortunate. Can we try creating the folder and launching it (just in case)?

**Reply by RobPasMue** [reply]
Also, would you consider using other Ansys CAD products like SpaceClaim? Performance might not be as high (in terms of speed), but it might be easier to get it working.

**Reply by ufsaa2022** [reply]
I tried it, but there's still a problem.

Running SCDM and Discovery using the script below works fine.

`from ansys.geometry.core import launch_modeler_with_discovery`
`modeler = launch_modeler_with_discovery()`

`from ansys.geometry.core import launch_modeler_with_spaceclaim`
`modeler = launch_modeler_with_spaceclaim()`

**Reply by ufsaa2022** [reply]
Which of the items below should I choose to install Geometry Service?

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/6f723455-b0f8-4008-ad95-b2ce40d75d1f)

**Reply by RobPasMue** [reply]
I believe it's the ``Design`` entry. However, we have been reported that sometimes the GeometryService does not get installed with 24R1. A solution I've come to see working is to install Ansys through the "Automated Installer" available in https://download.ansys.com/Current%20Release

**Reply by ufsaa2022** [reply]
It's already installed. Should I delete it and then install it?

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/9b742643-ceee-40ce-ad1e-22a618d9aa7f)

**Reply by RobPasMue** [reply]
If possible yes, let's give it a try... sorry that this is not working smoothly

**Reply by ufsaa2022** [reply]
Thanks :) I'll try it !

**Reply by ufsaa2022** [reply]
When I installed it with "Automated Installer", the GeometryService folder was automatically created.
However, it still doesn't run.
The same ConnectionError occurs.

**Reply by RobPasMue** [reply]
Okay good to know - we are one step further at least. Do you have a ``Logs`` folder now where you can look for server logs? If so could you share them with us?

**Reply by ufsaa2022** [reply]
Unfortunately the Logs folder still does not exist.
I tried creating a folder, but no files were created.

**Reply by RobPasMue** [reply]
That is once again unfortunate... Okay, so, next thing we are going to do is "brute force" launch the Geometry Service. In order to do this, you will have to search for the following file:

``C:\Program Files\ANSYS Inc\v241\GeometryService\Presentation.ApiServerDMS.exe``

Once you've found it, double click it, and you will get a batch terminal with a running Geometry service, similar to the following one:

![tempsnip](https://github.com/ansys/pyansys-geometry/assets/37798125/10b54f2a-1bf1-41b9-856c-ebcb0778dc15)

Could you share the logs with us in that case? Also, if everything seems fine, this means that you were able to run the Geometry Service. You could then connect to it using the ``Modeler()`` object instead of having to call ``launch_modeler...()``

**Reply by ufsaa2022** [reply]
When I just double-click, an error occurs, but when I right-click and run as administrator, the log is output as shown below.

![image](https://github.com/ansys/pyansys-geometry/assets/112993165/5dd3cc51-e0f6-4245-909a-4603f42b9fc2)

**Reply by RobPasMue** [reply]
Interesting! So, it seems like your user does not have the rights to run executable files by default, for some reason. This is probably what is causing the process to fail when requesting to launch. Something we can try is to open an admin terminal and launch your Python script - to see if it works :) want to give it a try?

**Reply by RobPasMue** [reply]
You will have to close the previous batch terminal as well!

**Reply by ufsaa2022** [reply]
WOW ! 
I'm using Visual Studio Code, and when I run as administrator, the script below runs fine.
I'm not sure why I need to run it as administrator.

`from ansys.geometry.core import launch_modeler_with_geometry_service`
`modeler = launch_modeler_with_geometry_service()`

**Reply by ufsaa2022** [reply]
And if "Presentation.ApiServerDMS.exe" is running as administrator, the script below also works.

`from ansys.geometry.core import Modeler`
`modeler = Modeler()`

**Reply by RobPasMue** [reply]
> And if "Presentation.ApiServerDMS.exe" is running as administrator, the script below also works.
> 
> `from ansys.geometry.core import Modeler` `modeler = Modeler()`

Yes, that script would work independently of the kind of terminal! That's because you are just connecting to the service and not launching it. Apparently, on your system, you seem to have some limitations on the kind of files you can run from a normal user terminal - but I'm glad we figured it out!

> WOW !
> I'm using Visual Studio Code, and when I run as administrator, the script below runs fine.
> I'm not sure why I need to run it as administrator.

PyAnsys Geometry, behind the scenes, is using subprocess to launch the Geometry Service for you... meaning that if your current terminal session does not have the needed rights, it will fail - just like it happen when you clicked it normally instead of running it as an admin.

I suggest you contact your IT administrator to solve this issue if possible! That way you won't need to run VSCode as an admin every timt you want to run it. =)

**Reply by ufsaa2022** [reply]
I'm so grateful for all your help !
Have a great day ! :)

**Reply by RobPasMue** [reply]
My pleasure! Feel free to post any issues/discussions with any other upcoming problems you have. Also, if you like our project, you can leave us a star on GitHub! I'm grateful you managed to get it working and sorry for all the forth and back. I hope the rest of your experience is smoother!

**Reply by ufsaa2022** [reply]
I left a star on GitHub ★☆★☆★
Thank you ~!

**Reply by RobPasMue** [reply]
Awesome!!!
Thanks a lot @ufsaa2022 - hope to hear from you soon!
